### PR TITLE
Late Pull Request

### PR DESCRIPTION
**Pull request by saleto**
Json parser looking good but i'm stuck with classAnnotation, please review my code

### CODE REVIEW

File: src/main/java/pb360/model/JsonDataModel.java [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
+
 import pb360.model.embedded.JsonNode;
+
 import com.fasterxml.jackson.annotation.JsonInclude.Include;
 
 @JsonInclude(Include.NON_DEFAULT)
 public class JsonDataModel {
 	private String title;
-	private List<JsonNode> nodeList;
+	private List<JsonNode> Model;
```
Comment (hoan-salt) [inline src/main/java/pb360/model/JsonDataModel.java]: this is wrong, we have only a listNode, listNode will contain Model,Validator,Options...

File: src/main/java/pb360/model/embedded/AnnotationInside.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	private String annotationEnds;
+	private String annotations;
+
+	@JsonIgnoreProperties(ignoreUnknown = true)
+	public AnnotationInside(@JsonProperty("annotationStarts") String annotationStarts,
+			@JsonProperty("annotationContent") List<String> annotationContent,
+			@JsonProperty("annotationEnds") String annotationEnds,
+			@JsonProperty("annotation") String annotations) {
+		
+		if(annotations instanceof String)
```
Comment (hoan-salt) [inline src/main/java/pb360/model/embedded/AnnotationInside.java]: what means of this code?

File: src/main/java/pb360/model/embedded/AnnotationParam.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package pb360.model.embedded;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AnnotationParam {
+	@JsonFormat(with = JsonFormat.Feature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
+	private String annotationParam;
```
Comment (hoan-salt) [inline src/main/java/pb360/model/embedded/AnnotationParam.java]: Why we need create a Class for containning one String variable?

File: src/main/java/pb360/model/embedded/Line.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package pb360.model.embedded;
+
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Line {
+	private String line;
```
Comment (hoan-salt) [inline src/main/java/pb360/model/embedded/Line.java]: I think we should not create a model for containning one variable

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 5]
```diff
@@ -94,40 +86,202 @@ private MessageObject getCurrentDateTime() {
...
-				stringBuilder.append(buffReader.readLine());
+
+			Object obj = parser.parse(new FileReader(link));
+
+			
+			JSONObject jsonObject = (JSONObject) obj;
+			//title
+			String title = (String) jsonObject.get("title");
+			// loop modelArray
+			//Model
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java]: the code looks fine, but we need separate model to a small function, validator to a small function, that means we can easy to maintain in future

File: src/main/java/pb360/model/embedded/Line.java (hunk 4 above)
Comment (saleto) [reply src/main/java/pb360/model/embedded/Line.java]: i thought i should use @JsonProperty for each line. I deleted this

File: src/main/java/pb360/model/embedded/AnnotationParam.java (hunk 3 above)
Comment (saleto) [reply src/main/java/pb360/model/embedded/AnnotationParam.java]: i thought i should use @jsonproperty for each line. I deleted this

File: src/main/java/pb360/model/embedded/AnnotationInside.java (hunk 2 above)
Comment (saleto) [reply src/main/java/pb360/model/embedded/AnnotationInside.java]: This code was to find out if annotations is 1 String or Array but it doesn't work so i delete it too

File: src/main/java/pb360/model/JsonDataModel.java (hunk 1 above)
Comment (saleto) [reply src/main/java/pb360/model/JsonDataModel.java]: I will change it to listNode and put the other in them

File: src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java [hunk 6]
```diff
@@ -94,40 +86,221 @@ private MessageObject getCurrentDateTime() {
...
+				}
+			}
+
+		}
+	}
+	
+	
+	
+
+	public void write2Java(String stringData, String fileName) {
```
Comment (hoan-salt) [inline src/main/java/pb360/service/impl/GenerateRestApiServiceImpl.java]: refactor function name; it should be writeToJavaFile

**Pull request merged**